rs/consensus/src/consensus/batch_delivery.rs

```diff
@@ -233,6 +233,7 @@ pub fn deliver_batches(
             time: block.context.time,
             consensus_responses,
             blockmaker_metrics,
+            replica_version: current_replica_version.clone(),
```

Comment:
We should use the replica version of the block here, as it is guaranteed to have gone through consensus.
```suggestion
            replica_version: block.version().clone(),
```